### ISSUE TITLE
Support for Prompt Blending / Prompt Alternating / Prompt Editing like in Automatic1111

### ISSUE THREAD

**Issue by Hellisotherpeople** [oobabooga/text-generation-webui#1127]
**Description**

Inputs to LLM consist of strings which are tokenized and these tokens are converted into embeddings which are the input that the LLM decodes and generates text against. 

Why don't we add support for allowing us to blend the embeddings of two or more tokens/words together? For example, this would enable things like "What's the definition of {orange|apple|pineapple:3}" where this is the embedding for the weighted average of the tokens in orange, apple, and 3x pineapple. 

We can also implement something of a ["prompt alternating" ](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Features#alternating-words) by changing the input prompt as shown at each timestep that the LLM generates, and we can implement ["prompt editing"](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Features#prompt-editing) in similar fashion

We can even implement the advanced token interpolation techniques introduced in prompt fusion. 

**Additional Context**

This is supported in automatic1111 through a [ custom script ](https://github.com/amotile/stable-diffusion-backend/tree/master/src/process/implementations/automatic1111_scripts) as well as in[ multiple extensions. ](https://github.com/ljleb/prompt-fusion-extension)

**Comment by mcmonkey4eva** [comment]
Not all concepts quite so directly map between SD and LLMs. For example, re `a ["prompt alternating" ](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Features#alternating-words)by changing the input prompt as shown at each timestep that the LLM generates` - well the LLM _doesn't have_ timesteps. That's a concept unique to the diffusion process. LLMs generate one token at a time linearly.

That said, there definitely *is* some potential for mucking with the inputs in beneficial ways. It's just, a lot more complicated.

**Comment by Hellisotherpeople** [comment]
> Not all concepts quite so directly map between SD and LLMs. For example, re `a prompt alternating by changing the input prompt as shown at each timestep that the LLM generates` - well the LLM _doesn't have_ timesteps. That's a concept unique to the diffusion process. LLMs generate one token at a time linearly. Similarly, things like weighting don't apply, as, uh, weighting doesn't exist.
> 
> That said, there definitely _is_ some potential for mucking with the inputs in beneficial ways. It's just, a lot more complicated.

No, they do map directly, you just need to think about it. I've thought about it, and have even published work in similar directions with LLMs. 

let's be clear, LLMs operate in an auto-regressive fashion by being prompted with an input, and then generating one token at a time as you mention. We then append the token it generates, and repeat the process. We can input prompt alternating by _alternating the previous input prompts between the user given two_

Imagine that you want to get 20 tokens of output with prompt alternating between two prompts: "I like apple" and "I like bananas", making the input prompt look like this: [I like apple:I like bananas]

For the first token generated, the input is "I like apple", let's say it generates "because"
For the second token generated, the input is "I like bananas because", let's say it generates "I" 
for the third token generated, the input is "I like apple because I" and it generates... 

Please in the future, trust that when I identify that a mapping exists between features in these spaces, that this mapping actually does exist.

**Comment by Hellisotherpeople** [comment]
> Not all concepts quite so directly map between SD and LLMs. For example, re `a prompt alternating by changing the input prompt as shown at each timestep that the LLM generates` - well the LLM _doesn't have_ timesteps. That's a concept unique to the diffusion process. LLMs generate one token at a time linearly. Similarly, things like weighting don't apply, as, uh, weighting doesn't exist.
> 
> That said, there definitely _is_ some potential for mucking with the inputs in beneficial ways. It's just, a lot more complicated.

Btw I wrote a github gist to prove all of what I say above and it went viral on HN

https://gist.github.com/Hellisotherpeople/45c619ee22aac6865ca4bb328eb58faf
